result-watcher.bal

```diff
@@ -300,79 +365,32 @@ public isolated class ResultWatcher {
-
-            // TODO: update progress in any case???
-            database:TimelineSubstep[]? resultSubsteps = result?.substeps;
-            database:Progress? progress = result?.progress;
-            string? taskLog = result?.taskLog;
-            if resultSubsteps != () {
-                transaction {
-                    database:TimelineSubstepSQL[] substeps = check database:getTimelineSubsteps(self.stepId);
-                    // cast db substeps to format in result
-                    database:TimelineSubstep[] tempSubsteps = [];
-                    foreach var substep in substeps {
-                        database:TimelineSubstep tempSubstep = {
-                            substepId: substep.substepId,
-                            href: substep.href,
-                            hrefUi: substep.hrefUi,
-                            cleared: substep.cleared
-                        };
-                        tempSubsteps.push(tempSubstep);
-                    }
-                    // find new/changed substeps
-                    database:TimelineSubstep[] newSubsteps = from var substep in tempSubsteps
-                        where tempSubsteps.indexOf(substep) == ()
-                        select substep;
-                    // add in db
-                    foreach var substep in newSubsteps {
-                        check database:updateTimelineSubstep(self.stepId, substep); // TODO: perhaps nicer to pass TimelineSubstep
-                    }
-
-                    // TODO: check that UI didn't delete substeps
-                    // TODO: neuer substep -> alle davor auf cleared setzen und optional warning/error
-                    // TODO: checken ob welche rausfliegen oder daziwschen welche eingefügt werden... -> error, hrefs vergleichen, substep
-                    // wenn substepid nicht gesetzt darf auch nicht nachträglich gesetzt werden... 
-
-                    // write progress into db - 
-                    if progress != () && newSubsteps.length() > 0 {
-                        check database:updateTimelineProgress(self.stepId, progress);
-                    }
-
-                    // save current task log
-                    if taskLog != () {
-                        check database:updateTimelineTaskLog(self.stepId, taskLog);
-                    }
-                    check commit;
-                } on fail error err {
-                    io:println(err);
-                }
-
-                do {
+            do {
+                ResultProcessor processor = new (result, self.experimentId, self.stepId, self.resultEndpoint);
+                boolean isChanged = check processor.processIntermediateResult();
+                if isChanged {
                     check self.unschedule();
-                    // TODO: Better way to reschedule this???
+                    // TODO: Probably needs to be changed in the future
```

Comment:
It should use the same intervall config as all other schedules (alternatively the last used schedule e.g.g the last parameters for the call to schedule). But this TODO is ok for this PR.

Reply:
Maybe, but I want to tackle the scheduling problem in another PR. This comment was just to let you know about this issue.
We may even build an actual registry object that does the scheduling and unscheduling.

Reply:
Ok, sounds good